package/kedro_viz/data_access/repositories/catalog.py

```diff
@@ -23,9 +24,14 @@ def get_catalog(self) -> DataCatalog:
     def set_catalog(self, value: DataCatalog):
         self._catalog = value
 
-    @staticmethod
-    def strip_encoding(dataset_name: str) -> str:
-        return dataset_name.split("@")[0]
+    def _validate_layers_for_transcoding(self, dataset_name, layer):
+        if TRANSCODING_SEPARATOR in dataset_name:
+            dataset_name = _strip_transcoding(dataset_name)
+            existing_layer = self._layers_mapping.get(dataset_name)
+            if existing_layer is not None and existing_layer != layer:
+                raise ValueError(f"Transcoded datasets should have the same layer. Mismatch found for: {dataset_name}")
+        return dataset_name, layer
+
```

Comment:
I know it's copied from the `kedro` codebase, but I found it weird that the validation function actually return something.

Is it easy to move the `strip_transcoding` outside and the validation only validate but not return anything?

Reply:
done !